### PR TITLE
Adding test infrastructure and tests for null semantics. 

### PR DESCRIPTION
**Pull request by maumar**
Tests are disabled, since the functionality is not there yet.

### CODE REVIEW

File: test/EntityFramework.Core.FunctionalTests/NullSemanticsQueryFixtureBase.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+{
+    public abstract class NullSemanticsQueryFixtureBase<TTestStore>
+        where TTestStore : TestStore
+    {
+        public abstract TTestStore CreateTestStore();
+
+        public abstract NullSemanticsContext CreateContext(TTestStore testStore);
+
+        protected virtual void OnModelCreating(ModelBuilder modelBuilder)
+        {
```
Comment (anpete) [inline test/EntityFramework.Core.FunctionalTests/NullSemanticsQueryFixtureBase.cs]: Is all of this config required? Isn't Code First doing a bunch of this?

**Comment by anpete** [comment]
:shipit:
